### DISCUSSION TITLE
Which pin number (GPIO) to use on ESP32

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#25] [Q&A] [answered]
First off, thank you so much for this great project. 

Now, here is my question: Instead of using an ESP8266, I have on hand an ESP32 (HiLetgo ESP-WROOM-32). I set up all the electronics and tested my configuration for correct voltages. However, before connecting it to my alarm system (Vista 21IP), I would like to confirm the Pin numbers for the ESP32. 

Looking online, the default SDA/SCL to connect I2C device is GPIO 21 (SDA) and GPIO 22 (SCL) for the ESP32. While looking at the vistaAlarm.h code, I see the following: 
`
//esp32 use pins 4,13,16-39 
#ifdef ESP32
#define D1 18
#define D2 19
#define D5 21
#endif
`
Should these be D1 at 22 and D2 at 21? When I compare the number to the ESP8266, the GPIO matches the SDA/SCL pins. 

Now for D5, what should I use?

Now, looking at the vista.h code, there is again the definition for the D1, D3, D4, etc for the ESP8266 but no redefinition for the ESP32. Is that expected? Seems that we should also define the ESP32? 

As you see, I am in now way an expert here so your help is appreciated. Would it be possible to get an alternate schematics for the ESP32 showing the pins?

Thanks

**Comment by Dilbert66** [comment, marked as answer]
You can use any pin you want provided that none have a boot input level requirement or set boot output level. I use d1=18, d2=19 and d5=19.  You can use d1=22, d2=21 if you wish.    I use the scl/sda pins because that fit my needs and I am not using those for i2c comms.    

No, I do not have a schematic for an esp32 since really, it would not be much different than the one for the esp8266.   It's a very basic circuit.   As to the other defines in the vista.h file, those get overridden by the vistaalarm.h values.

**Reply by AlexandreUSA** [reply]
Thank you so much. I will use the same PIN numbers as you and see how it goes. Glad I did not have to match the exact same type of port as on the esp8266. That makes it pretty simple. Great work.
